spec/jstests-spec.js

```diff
@@ -195,3 +195,75 @@ describe("Initializing tooltips", function() {
   });
 });
 
+describe("Pressing next and previous buttons associated with a graph", function() {
+  it("generates correct fetch request when next/previous buttons are pressed", function() {
+
+    const prevButton = document.createElement("button");
+    prevButton.id = "prevbutton";
+    prevButton.onclick = () => navigateGraph(-1);
+    const nextButton = document.createElement("button");
+    nextButton.id = "nextbutton";
+    nextButton.onclick = () => navigateGraph(1);
+    document.body.appendChild(prevButton);
+    document.body.appendChild(nextButton);
+
+    expect(currGraphNum).toBe(0);
+    expect(numMutations).toBe(3);
+
+    nextButton.click();
+    expect(currGraphNum).toBe(1);
+
+    nextButton.click();
+    expect(currGraphNum).toBe(2);
+
+    nextButton.click();
+    expect(currGraphNum).toBe(3);
+
+    nextButton.click();
+    expect(currGraphNum).toBe(3);
+
+    expect(nextButton.disabled).toBe(true);
+    expect(prevButton.disabled).toBe(false);
+
+    prevButton.click();
+    expect(currGraphNum).toBe(2);
+    expect(nextButton.disabled).toBe(false);
+    expect(prevButton.disabled).toBe(false);
+
+    prevButton.click(); 
+    expect(currGraphNum).toBe(1);
+    expect(nextButton.disabled).toBe(false);
+    expect(prevButton.disabled).toBe(false);
+
+    nextButton.click();
+    expect(currGraphNum).toBe(2);
+
+    prevButton.click();
+    expect(currGraphNum).toBe(1);
+
+    prevButton.click();
+    expect(currGraphNum).toBe(0);
+
+    prevButton.click();
+    expect(currGraphNum).toBe(0);
+
+    prevButton.click();
+    expect(currGraphNum).toBe(0);
+    expect(nextButton.disabled).toBe(false);
+    expect(prevButton.disabled).toBe(true);
+ 
+    nextButton.click();
+    expect(currGraphNum).toBe(1);
+    expect(nextButton.disabled).toBe(false);
+    expect(prevButton.disabled).toBe(false);
+
+    const requestString = getUrl();
```

Comment:
Asserting that the correct mutationNum is passed should be its own test.

Reply:
moved everything under that line into a new test!